src/pip/_internal/utils/encoding.py

```diff
@@ -35,8 +32,10 @@ def auto_decode(data):
     # Lets check the first two lines as in PEP263
     for line in data.split(b'\n')[:2]:
         if line[0:1] == b'#' and ENCODING_RE.search(line):
-            encoding = ENCODING_RE.search(line).groups()[0].decode('ascii')
-            return data.decode(encoding)
+            result = ENCODING_RE.search(line)
+            if result:
```

Comment:
I'd `assert result` to stay closer to existing code semantics.

Reply:
Added `assert` in https://github.com/pypa/pip/pull/8277/commits/50cbd6a03245d6315da74c470cf1c481ef554cba